### PR TITLE
Make autoscalinggroup names unique to avoid duplicate asgs being created

### PR DESCRIPTION
**Pull request by humivo**
**Description:** Changed the asg naming to `name_prefix` which will take the prefix and add create a unique asg name afterwards.

**Testing:** Local testing passed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: terraform/ecs/main.tf [hunk 1]
```diff
@@ -126,7 +126,7 @@ resource "aws_launch_template" "launchtemp" {
 }
 
 resource "aws_autoscaling_group" "clusterasg" {
-  name                = "clusterasg"
+  name_prefix         = "clusterasg"
```
Comment (bryan-aguilar) [inline terraform/ecs/main.tf]: Can we use the testing id here or at least tag the auto scaling group with the testing ID?

Comment (humivo) [reply terraform/ecs/main.tf]: Added it and changed the other names

File: terraform/ecs/main.tf [hunk 2]
```diff
@@ -126,11 +126,23 @@ resource "aws_launch_template" "launchtemp" {
 }
 
 resource "aws_autoscaling_group" "clusterasg" {
-  name                = "clusterasg"
+  name_prefix         = "clusterasg-${module.common.testing_id}-"
```
Comment (rapphil) [inline terraform/ecs/main.tf:129]: should the name be different? https://github.com/aws-observability/aws-otel-collector/blob/main/tools/workflow/cleaner/autoscaling/clean.go#L29

File: terraform/ecs/main.tf [hunk 3]
```diff
@@ -126,11 +126,23 @@ resource "aws_launch_template" "launchtemp" {
...
   vpc_zone_identifier = module.basic_components.aoc_private_subnet_ids
   desired_capacity    = 1
   max_size            = 10
   min_size            = 1
+  tags = [
+    {
+      "key"                 = "ephermeral"
```
Comment (bryan-aguilar) [inline terraform/ecs/main.tf]:
```suggestion
      "key"                 = "ephemeral"
```

**Review by bryan-aguilar — APPROVED** [review]

**Review by rapphil — APPROVED** [review]

**Pull request merged**